app/logic/Mail/ActiveSync/ActiveSyncAccount.ts

```diff
@@ -454,35 +451,33 @@ export class ActiveSyncAccount extends MailAccount {
             break;
           case FolderType.Contacts:
           case FolderType.UserContacts:
-            let haveAddressbook = appGlobal.addressbooks.find(addressbook => addressbook.mainAccount == this);
-            let isMainAddressbook = !haveAddressbook; // TODO Determine default addressbook
-            if (!haveAddressbook) {
-              let addressbook = newAddressbookForProtocol("addressbook-activesync") as ActiveSyncAddressbook;
+            let isMainAddressbook = change.Type == FolderType.Contacts;
+            let addressbook = appGlobal.addressbooks.find((addressbook: ActiveSyncAddressbook) => addressbook.mainAccount == this && addressbook.serverID == change.ServerId) as ActiveSyncAddressbook | null;
+            if (!addressbook) {
+              addressbook = newAddressbookForProtocol("addressbook-activesync") as ActiveSyncAddressbook;
               addressbook.initFromMainAccount(this);
-              if (!isMainAddressbook) {
-                addressbook.name = sanitize.nonemptylabel(change.DisplayName, addressbook.name);
-              }
-              addressbook.url = sanitize.url(folderURL);
               addressbook.serverID = sanitize.nonemptystring(change.ServerId);
-              await addressbook.save();
-              appGlobal.addressbooks.add(addressbook);
             }
+            if (!isMainAddressbook) {
+              addressbook.name = sanitize.nonemptylabel(change.DisplayName, addressbook.name);
+            }
+            await addressbook.save();
```

Comment:
Why do we need to save the addressbook on every login?

Calendars and address books should be saved in the DB, including their name, and re-loaded on every startup. They should not need to be created, not need to be saved (unless something actually changed), and not need to be added to `appGlobal.addressbooks`, because it's already in there since load, long before we get there.